lms/djangoapps/certificates/views/support.py

```diff
@@ -184,37 +184,35 @@ def regenerate_certificate_for_user(request):
     if response is not None:
         return response
 
-    # Check that the course exists
-    course = modulestore().get_course(params["course_key"])
-    if course is None:
-        msg = _("The course {course_key} does not exist").format(course_key=params["course_key"])
+    user = params["user"]
+    course_key = params["course_key"]
+
+    try:
+        get_course_overview(course_key)
+    except CourseOverview.DoesNotExist:
+        msg = _("The course {course_key} does not exist").format(course_key=course_key)
         return HttpResponseBadRequest(msg)
 
     # Check that the user is enrolled in the course
-    if not CourseEnrollment.is_enrolled(params["user"], params["course_key"]):
-        msg = _("User {username} is not enrolled in the course {course_key}").format(
-            username=params["user"].username,
-            course_key=params["course_key"]
+    if not CourseEnrollment.is_enrolled(user, course_key):
+        msg = _("User {user_id} is not enrolled in the course {course_key}").format(
+            user_id=user.id,
```

Comment:
Question: if this is only called by the support tools, which already has access to _username_, should we keep the current behavior of returning the username?

Then again it could be called by other things in the future, so _id_ is a safer choice.

Reply:
This is a good question. I saw username being used and automatically updated it. I think _id_ is the safer choice as well. If we get any complaints then I think it will be an easy piece to revert.